src/FormRuntime.php

```diff
@@ -75,7 +75,7 @@ public function run(string $formName = '', bool $warn = true)
 
         $form->handleRequest($this->request);
 
-        if ($form->isSubmitted()) {
+        if ($form->isSubmitted() && $form->isValid()) {
```

Comment:
I was confused by this when I started implementing the validation, as I couldn't see where the data was actually validated. My worry was that it would be firing the post-submit event to handlers before the data (and therefore the CAPTCHA) was validated.

However this is unfounded - removing the `isValid()` call here still fires the validation and prevents the form being submitted.

Reply:
👍  Nice one! And I am on the same page with you, I was also confused the first time I came across this. Perhaps in a future version we can differentiate between `PostSubmit` and `"PostValidatedSubmit"` 🤷‍♂️